tests/test_cpu_models.py

```diff
@@ -359,5 +357,63 @@ def test_single_gpu_model(tmpdir):
     tutils.run_model_test(trainer_options, model)
 
 
+def test_nan_loss_detection(tmpdir):
+    test_step = 8
+
+    class InfLossModel(LightTrainDataloader, TestModelBase):
+
+        def training_step(self, batch, batch_idx):
+            output = super().training_step(batch, batch_idx)
+            if batch_idx == test_step:
+                if isinstance(output, dict):
+                    output['loss'] /= 0  # make loss infinite
```

Comment:
here may cry about division by 0, so rather set it as Inf directly

Reply:
ok, thanks!